plugins/webp-uploads/helper.php

```diff
@@ -348,3 +354,32 @@ function webp_uploads_should_discard_additional_image_file( array $original, arr
 	}
 	return false;
 }
+
+/**
+ * Checks if a mime type is supported by the server.
+ *
+ * Includes special handling for false positives on AVIF support.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $mime_type The mime type to check.
+ * @return bool Whether the server supports a given mime type.
+ */
+function webp_uploads_mime_type_supported( string $mime_type ): bool {
+	if ( ! wp_image_editor_supports( array( 'mime_type' => $mime_type ) ) ) {
+		return false;
+	}
+
+	// In certain server environments Image editors can report a false positive for AVIF support.
+	if ( 'image/avif' === $mime_type ) {
+		$editor = _wp_image_editor_choose( 'image/avif' );
+		if ( 'WP_Image_Editor_GD' === $editor ) {
```

Comment:
And it seems `_wp_image_editor_choose` returns an instance of the object, not the class. So should this rather be:

```suggestion
		if ( $editor instanceof WP_Image_Editor_GD ) {
```

Or why `WP_Image_Editor_GD` specifically and not `WP_Image_Editor`?

Reply:
Ah, because of the `function_exists()` check.

Reply:
note `_wp_image_editor_choose` returns a string

Reply:
Oops. I must have looked at the wrong spot. In that case, how about:

```suggestion
		if ( is_a( $editor, WP_Image_Editor_GD::class, true ) ) {
```

This will support subclasses of `WP_Image_Editor_GD` being able to be used here.

Reply:
For example, with Dominant Color Images active, this could be `Dominant_Color_Image_Editor_GD` or `Dominant_Color_Image_Editor_Imagick`.

Reply:
> For example, with Dominant Color Images active, this could be Dominant_Color_Image_Editor_GD or Dominant_Color_Image_Editor_Imagick.

Ah right, good point.

> 		`if ( is_a( $editor, WP_Image_Editor_GD::class, true ) ) {`

I like this approach. Do you mean to have $editor as the second argument and the class as the first argument though? https://www.php.net/manual/en/function.is-a.php

Reply:
Oh, NM I was reading that wrong - this isgood. would be nice to have some tests for the case you described though, `Dominant_Color_Image_Editor_Imagick`